docs/how_to_publish_to_cmr_with_earthdata_varinfo.ipynb

```diff
@@ -0,0 +1,314 @@
+    "from varinfo import VarInfoFromNetCDF4\n",
+    "\n",
+    "# Import functions for creating UMM-Var entry and\n",
+    "# publishing a UMM-Var entry\n",
+    "from varinfo.umm_var import (get_all_umm_var, get_umm_var, publish_all_umm_var,\n",
+    "                             publish_umm_var)\n",
+    "# Singular function call\n",
+    "from varinfo.generate_umm_var import generate_collection_umm_var"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Update authorization header to include your bearer token (e.g. 'Bearer TOKEN')\n",
+    "# or LaunchPad token (e.g. TOKEN)\n",
+    "auth_header = 'Header + Token or Only token'"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get granule response and granule download URL with ```get_granules``` and ```get_granule_link``` for collection **M2I1NXASM**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "granule_response = get_granules(concept_id='C1256535511-EEDTEST',\n",
+    "                                cmr_env=CMR_UAT,\n",
+    "                                auth_header=auth_header)\n",
+    "link = get_granule_link(granule_response)\n",
+    "link"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Then download the granule locally with `download_granule`"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Saves granule to current directory or\n",
+    "# out_directory=/usr/given/path\n",
+    "download_granule(link, auth_header=auth_header)"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Instantiate a ```VarInfoFromNetCDF4``` object for a local NetCDF-4 file. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Instantiate a VarInfoFromNetCDF4 object for a local NetCDF-4 file.\n",
+    "var_info = VarInfoFromNetCDF4('MERRA2_400.inst1_2d_asm_Nx.20220130.nc4',\n",
+    "                              short_name='M2I1NXASM')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Retrieve a dictionary of UMM-Var JSON records"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Keys are the full variable paths, values are \n",
+    "# UMM-Var schema-compatible, JSON-serialisable dictionaries.\n",
+    "umm_var_dict = get_all_umm_var(var_info)\n",
+    "umm_var_dict"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Publish all the UMM-Var entries for M2I1NXASM"
+   ]
```

Comment:
What should the user expect to see when this code is completed?

Reply:
* [ ] Add example of return code